src/utils/documentUtils.ts

```diff
@@ -40,4 +43,44 @@ const getDocPropsProperties = async (zip: JSZip): Promise<{ doc: Document; prope
     return { doc, properties };
 };
 
-export default { createOrUpdateProperty, getDocPropsProperties };
+const getCellReference = (col: number, row: number): string => {
+    return String.fromCharCode(col + A) + "$" + (row).toString();
+} 
+
+const createCell = (doc: Document, parent: Element, colIndex: number, rowIndex: number, dataType: number, data: string): void => {
+    const cell = doc.createElementNS(doc.documentElement.namespaceURI, "c");
+    cell.setAttribute("r", getCellReference(colIndex, rowIndex + 1).replace("$", ''));
+    const cellData = doc.createElementNS(doc.documentElement.namespaceURI, "v");
+    updateCellData(dataType, data , cell, cellData);
+    cell.appendChild(cellData);
+    parent.appendChild(cell);
+    colIndex++;
+}
+
+const updateCellData = (dataType: number, data: string, newCell: Element, cellData: Element) => {
+    if (dataType == dataTypes.string) {
+        newCell.setAttribute("t", "str");
+        cellData.textContent = data;
```

Comment:
textContent can be set once at the end of the function, and in case you want to override given value (errors) - override in you if/else statements.
This is for now, before we consider changing the approach to be more typed.

Reply:
Also - if/else pattern is a bit complex here without cause, either do if / else if / etc or do a switch.